frontend/server/src/main/java/org/pytorch/serve/wlm/ModelManager.java

```diff
@@ -205,81 +209,168 @@ private ModelArchive createModelArchive(
         return archive;
     }
 
+    private void setupModelVenv(Model model)
+            throws IOException, InterruptedException, ModelException {
+        if (!model.isUseVenv()) {
+            return;
+        }
+
+        File venvPath = EnvironmentUtils.getPythonVenvPath(model);
+        List<String> commandParts = new ArrayList<>();
+        commandParts.add(configManager.getPythonExecutable());
+        commandParts.add("-m");
+        commandParts.add("venv");
+        commandParts.add("--clear");
+        commandParts.add("--system-site-packages");
+        commandParts.add(venvPath.toString());
+
+        ProcessBuilder processBuilder = new ProcessBuilder(commandParts);
+
+        if (isValidDependencyPath(venvPath)) {
+            processBuilder.directory(venvPath.getParentFile());
+        } else {
+            throw new ModelException(
+                    "Invalid python venv path for model "
+                            + model.getModelName()
+                            + ": "
+                            + venvPath.toString());
+        }
+        Map<String, String> environment = processBuilder.environment();
+        String[] envp =
+                EnvironmentUtils.getEnvString(
+                        configManager.getModelServerHome(),
+                        model.getModelDir().getAbsolutePath(),
+                        null);
+        for (String envVar : envp) {
+            String[] parts = envVar.split("=", 2);
```

Comment:
What is 2 here?

Reply:
It indicates the number parts to split the string into using regex. This is shown in the docs: https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#split-java.lang.String-int-

I reused this existing logic from here:
https://github.com/pytorch/serve/blob/7c4559c55835c7848cac0d5f67ca1016fd2c3ce4/frontend/server/src/main/java/org/pytorch/serve/wlm/ModelManager.java#L251-L257